app/src/main/java/com/bogdan/codeforceswatcher/features/news/NewsFragment.kt

```diff
@@ -49,61 +115,112 @@ class NewsFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener, StoreSubs
     }
 
     override fun onNewState(state: NewsState) {
-        if (state.status == NewsState.Status.PENDING) {
-            swipeRefreshLayout.isRefreshing = true
-        } else {
-            swipeRefreshLayout.isRefreshing = false
-            val items = mutableListOf<NewsItem>()
-
-            val feedbackController = FeedbackController.get()
-
-            if (feedbackController.shouldShowFeedbackCell()) {
-                items.add(NewsItem.FeedbackItem(feedbackController.feedUIModel))
-                newsAdapter.callback = {
-                    onNewState(state)
-                }
-            }
+        val items = mutableListOf<NewsItem>()
 
-            val news = state.news.filter {
-                return@filter if (it is News.PinnedPost) settings.readLastPinnedPostLink() != it.link else true
-            }
+        val feedbackController = FeedbackController.get()
```

Comment:
Can't we move it to the common module / Redux?

Reply:
I looked it up, `settings` store user settings, on iOS and Android it is implemented using system, so I as understand it - no.

Reply:
But if we store it in another way - I think yes.

Reply:
OK, I see, there is work to do. Can you create a refactoring ticket "Move FeedbackController to the common module"? As for Settings, there are multiple libraries that encapsulate differences between platforms and allow using them from the common module. It also can be done manually.